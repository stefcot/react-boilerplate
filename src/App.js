import React from 'react'

class App extends React.Component {
  render() {
    return (
      <main>
        <h1>Base for your project</h1>
      </main>
    )
  }
}

export default App
