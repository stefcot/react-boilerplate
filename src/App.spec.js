// Initialize a set of tests
import React from 'react'
import { render, cleanup } from '@testing-library/react'
import App from './App'

const setup = () => {
  const { container, getByText, asFragment } = render(<App />)
  return { container, getByText, asFragment }
}

describe('App', () => {
  afterEach(cleanup)
  // Supposed to fail cause of the lazy loading,
  // Otherwise, install babel-plugin-dynamic-import-node
  // cause jest runs on node
  it('It should render App component', () => {
    render(<App />)
  })

  it('It should render App component the same as snapshot', () => {
    const { asFragment } = setup()

    expect(asFragment()).toMatchSnapshot()

    // For state testing and others functionnal assertions
    // It's better to use enzyme
  })
})
