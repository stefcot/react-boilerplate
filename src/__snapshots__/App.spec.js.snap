// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App It should render App component the same as snapshot 1`] = `
<DocumentFragment>
  <main>
    <h1>
      Base for your project
    </h1>
  </main>
</DocumentFragment>
`;
